Userland/Utilities/wasm.cpp

```diff
@@ -351,6 +372,34 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     if (attempt_instantiate) {
         Wasm::AbstractMachine machine;
+        Optional<Wasm::Wasi::Implementation> wasi_impl;
+
+        if (wasi) {
+            wasi_impl.emplace(Wasm::Wasi::Implementation::Details {
+                .provide_arguments = [&] {
+                    Vector<String> strings;
+                    for (auto& string : args_if_wasi)
+                        strings.append(String::from_utf8(string).release_value_but_fixme_should_propagate_errors());
+                    return strings; },
+                .provide_environment = {},
+                .provide_preopened_directories = [&] {
+                    Vector<Wasm::Wasi::Implementation::MappedPath> paths;
+                    for (auto& string : wasi_preopened_mappings) {
+                        auto split_index = string.find(':');
+                        if (split_index.has_value()) {
+                            LexicalPath host_path { Core::DeprecatedFile::real_path_for(string.substring_view(0, *split_index)) };
```

Comment:
does LibFilesystem not have this API yet? It would be preferred to use that over adding more uses of DeprecatedFile.

Reply:
LibFileSystem's APIs use the new String, LexicalPath and Core::System use DeprecatedString - going through LibFileSystem is gonna make us do DeprecatedString -> String -> DeprecatedString.
I'd rather just keep the DeprecatedFile and not pretend this is using DeprecatedStuff.